lib/src/bridge/characteristics_mixin.dart

```diff
@@ -20,10 +20,16 @@ mixin CharacteristicsMixin on FlutterBLE {
           )
           .catchError((errorJson) =>
               Future.error(BleError.fromJson(jsonDecode(errorJson.details))))
-          .then((rawJsonValue) =>
-              _parseCharacteristicWithValueWithTransactionIdResponse(
-                      peripheral, rawJsonValue)
-                  .value);
+          .then((rawValue) {
+            String rawJsonValue = "";
+            if (rawValue is String) {
```

Comment:
This can actually be get rid of by changing to `.invokeMethod<String>(...)`

Reply:
yeah, so I figured.  As later I went and explored invokeMethod nullsafe amendments..
I did those on the latest commit that ended up being reverted. :)
we can refactor the whole code with this making it much more type correct! ;)